src/Stratis.SmartContracts.Executor.Reflection/ContractLogging/ContractLogHolder.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using NBitcoin;
+using Stratis.SmartContracts.Core;
+using Stratis.SmartContracts.Core.Receipts;
+using Stratis.SmartContracts.Executor.Reflection.Serialization;
+
+namespace Stratis.SmartContracts.Executor.Reflection.ContractLogging
+{
+    public class ContractLogHolder : IContractLogger
+    {
+        private readonly IContractPrimitiveSerializer serializer;
+        private readonly Network network;
+        private readonly List<RawLog> rawLogs;
+
+        public ContractLogHolder(IContractPrimitiveSerializer serializer, Network network)
+        {
+            this.serializer = serializer;
+            this.network = network;
+            this.rawLogs = new List<RawLog>();
+        }
+
+        public void Log<T>(ISmartContractState smartContractState, T toLog)
+        {
+            this.rawLogs.Add(new RawLog(smartContractState.Message.ContractAddress.ToUint160(this.network), toLog));
```

Comment:
Because ContractLogHolder is local to the VM create/call methods, could we pass the address in the constructor and avoid the need to pass in the entire state? Will make testing slightly nicer as well.